lib/new_relic/agent.rb

```diff
@@ -605,6 +605,26 @@ def add_custom_span_attributes(params)
       end
     end
 
+    # Set the user id for the current transaction. When present, this value will be included in the agent attributes for transaction and error events as 'enduser.id'.
+    #
+    # @param [String] user_id    The user id to add to the current transaction attributes
+    #
+    # @api public
+    def set_user_id(user_id)
+      record_api_supportability_metric(:set_user_id)
```

Comment:
Given that we expect that `set_user_id` will be called very frequently (at least once per transaction), I think incrementing can end up creating a lot of egress. Perhaps we should simply record once, so that we know the feature is being used without concern for how many invocations.

```diff
+ SET_USER_ID_SUPPORTABILITY_METRIC = "#{NewRelic::SupportabilityHelper::API_SUPPORTABILITY_PREFIX}set_user_id/Invoked"


- record_api_supportability_metric(:set_user_id)
+ record_metric_once(SET_USER_ID_SUPPORTABILITY_METRIC)
```

Reply:
That makes sense, my concern with that would be it would cause inconsistencies with the way we record this metric compared to all other api method metrics. We have a lot of other API methods that can also be called once or more per transaction, so this method wouldn't be doing anything unusually time consuming as is.
If this is a change we want to make, I think we should make this change to all API methods to keep things consistent.

Reply:
Great call on consistency, @tannalynn. We discussed this one more offline and will keep the original approach here as-is.